File: backend/src/main/java/com/example/Flicktionary/domain/series/entity/SeriesCast.java
package com.example.Flicktionary.domain.series.entity;

import com.example.Flicktionary.domain.actor.entity.Actor;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SeriesCast {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "series_id", nullable = false)
    private Series series;

    @ManyToOne
    @JoinColumn(name = "actor_id", nullable = false)
    private Actor actor;

    private String characterName;
}

File: backend/src/main/kotlin/com/example/Flicktionary/domain/series/entity/SeriesCast.kt
package com.example.Flicktionary.domain.series.entity

import com.example.Flicktionary.domain.actor.entity.Actor
import jakarta.persistence.*

@Entity
class SeriesCast(
    @ManyToOne
    @JoinColumn(name = "series_id", nullable = false)
    val series: Series,
    @ManyToOne
    @JoinColumn(name = "actor_id", nullable = false)
    val actor: Actor,
    @Column(length = 500)
    val characterName: String = ""
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private var _id: Long? = null;

    var id: Long
        get() = _id ?: 0
        set(value) {
            _id = value
        }
}
